### ISSUE TITLE
Deactivate cloud synchronization

### ISSUE THREAD

**Issue by dextreem** [Estimote/Android-Proximity-SDK#11]
## Prerequisites

* [X ] Put an X between the brackets on this line if you have done all of the following:
    * My beacons have Estimote Monitoring enabled (Check it in the Estimote Cloud, or via app)
    * My Android device/devices supports BLE and has Android OS version >= 5.0.0 
    * My Android device/devices have bluetooth enabled
    * My app has Location Permissions granted
    
## Basic information

**Estimote SDK version:** `0.1.0-alpha.6`

**Android devices affected:** All devices

**Android OS version affected:** All versions

**Beacon hardware version:** Doesn't matter

## Description

**How can I deactivate the cloud synchronization?**

Hi all,

this is not an issue, but more a regular question:
In my scenario I want to track users, i.e. I need to know **who** was at which location at what time? I could not find any request achieving this in the cloud API documentation ([docs](https://cloud.estimote.com/docs)). Does anybody know if there is a (maybe hidden) option for this?

The other alternative that I have is to setup a simple REST API by myself. Therefore I'd like to deactivate the **send-to-cloud**-option that seems to be activated per default:

`12-31 14:11:55.918 26771-26771/com.dextreem.gs D/ProximityAnalyticsSink: Successfully sent events to cloud.`

The idea behind this request is simply to save some data usage (and battery usage) I don't need. Is there an option for this?

## The code I use

I use the same code as described in the [repo](https://github.com/Estimote/Android-Proximity-SDK).

**Comment by pawelDylag** [comment]
Hey @dextreem 

As for unnecessary logs - they are turned on for the `alpha` version , but we will disable them by default later on. 

As for using cloud connection - our SDK needs it to resolve `attachments` to physical device identifiers, so for now there is no possibility to use it without internet connections (will be possible in the future releases to use only plain identifiers, without `attachments`).

In addition, our SDK reports `analytics data` to our cloud, so that you can see how many visits are in your beacons' zones. It is enabled by default, but you can simply turn it off like this:

``` Kotlin
   proximityObserver = ProximityObserverBuilder(applicationContext, cloudCredentials)
                .withAnalyticsReportingDisabled()
                .build()
```
...just simply add `.withAnalyticsReportingDisabled()` when building your `ProximityObserver`.

Regards, 
Paweł

**Comment by dextreem** [comment]
Hi Paweł,

thank you very much for your answer, that helped me a lot! :-)

> As for using cloud connection - our SDK needs it to resolve attachments to physical device identifiers, so for now there is no possibility to use it without internet connection (will be possible in future releases to use only plain identifiers, without using attachments).

I already observed that. In fact, my app crashes at startup when I don't have internet connection. Is it true that the SDK needs connection while startup **only**, i.e. after startup the mapping from assignment to beacon can be done locally? Why I'm asking: When scanning in background mode (and a connection is potentially not available) I obviously still need to log when entering/exiting (which) beacon region.

Can you say when the plain-identifier feature will be added?

Again, thank you very much.

Best regards
Dominic

**Comment by pawelDylag** [comment]
Hey @dextreem 

Yes, the mapping is done only at the startup. Also, we will handle `no internet connection` cases soon, so that you won't get crashes. 

The plain identifier feature is being discussed, but for sure it won't be introduced in the next release.

Also, we were thinking about extracting a method to prefetch data earlier, and start scanning in some time in the future.

Something like this: 

``` Kotlin
 
proximityObserver = ProximityObserverBuilder(applicationContext, cloudCredentials)
                .withBalancedPowerMode()
                .build()
       
val venueZone = proximityObserver.zoneBuilder()
                .forAttachmentKeyAndValue("venue", "myshop")
                .inFarRange()
                .withOnEnterAction(makeVenueFilled)
                .withOnExitAction(makeVenueWhite)
                .create()

proximityObserver
                .addProximityZone(venueZone)
                .prefetchCloudData()
...
// and then somewhere in the future
 observationHandler = proximityObserver.start()
```
Tell me what you think about it - would that be useful to you?

Regards, 
Paweł

**Comment by dextreem** [comment]
Hi Paweł,

thanks again for your answer.

To your proposal: This would be indeed very helpful in handling events as "no internet connection at startup". But when you plan on handling these cases on SDK layer, this would be obsolete, wouldn't it?

What do you think about the following approach: Instead of a **prefetch** allow the user (developer) a **fetch** at any time and additionally provide a method to store the assignment locally (on the devices store). In my case, the beacons won't change over time, so it suffices to get the assignment once and from then one just use this one. For users that have potentially changing beacons, a periodically `fetch()` or similar would do the trick.

This approach would be close to yours, with the difference that the user can fetch at any time and the fetched assignment is stored locally for future processing.

Thanks and best regards
Dominic

**Comment by dextreem** [comment]
Hi @pawelDylag,

any news to this case? I've seen that the new alpha version was released. The `no internet connection` case was not handled here, was it?

Thanks and best regards
Dominic

**Comment by pawelDylag** [comment]
Hey @dextreem 

Sorry for the delay - we're totally busy here working on the new stuff :) 
We have discussed the possibility of fetching the data separately. We agreed that this will be useful and it will be introduced before version `1.0.0` for sure. We've got it scheduled in our backlog, but for now we need to do higher priority stuff. I'll let you know when we will implement this feature :)

Also we've got some other ideas on how to implement `offline` functionality, but I will keep you informed on the way. 

Regards, 
Paweł
